connector/src/main/scala/com/microsoft/kusto/spark/datasource/KustoFilter.scala

```diff
@@ -0,0 +1,122 @@
+      case EqualNullSafe(attr, value) if value == null => unaryScalarOperatorFilter(attr, "isnull")
+      case EqualNullSafe(attr, value) => binaryScalarOperatorFilter(schema, attr, value, "==")
+      case GreaterThan(attr, value) => binaryScalarOperatorFilter(schema, attr, value, ">")
+      case GreaterThanOrEqual(attr, value) => binaryScalarOperatorFilter(schema, attr, value, ">=")
+      case LessThan(attr, value) => binaryScalarOperatorFilter(schema, attr, value, "<")
+      case LessThanOrEqual(attr, value) => binaryScalarOperatorFilter(schema, attr, value, "<=")
+      case In(attr, values) => unaryOperatorOnValueSetFilter(schema, attr, values, "in")
+      case IsNull(attr) => unaryScalarOperatorFilter(attr, "isnull")
+      case IsNotNull(attr) => unaryScalarOperatorFilter(attr, "isnotnull")
+      case And(left, right) => binaryLogicalOperatorFilter(schema, left, right, "and")
+      case Or(left, right) => binaryLogicalOperatorFilter(schema, left, right, "or")
+      case Not(child) => unaryLogicalOperatorFilter(schema, child, "not")
+      case StringStartsWith(attr, value) => stringOperatorFilter(schema, attr, value, "startswith_cs")
+      case StringEndsWith(attr, value) => stringOperatorFilter(schema, attr, value, "endswith_cs")
+      case StringContains(attr, value) => stringOperatorFilter(schema, attr, value, "contains_cs")
+      case _ => None
+    }
+  }
+
+  private def binaryScalarOperatorFilter(schema: StructType, attr: String, value: Any, operator: String): Option[String] = {
+    getType(schema, attr).map {
+      dataType => s"$attr $operator ${format(value, dataType)}"
+    }
+  }
+
+  private def unaryScalarOperatorFilter(attr: String, function: String): Option[String] = {
+    Some(s"$function($attr)")
+  }
+
+  private def binaryLogicalOperatorFilter(schema: StructType, leftFilter: Filter, rightFilter: Filter, operator: String): Option[String] = {
+    val left = buildFilterExpression(schema, leftFilter)
+    val right = if(left.isEmpty) None else buildFilterExpression(schema, rightFilter)
+
+    if (left.isEmpty || right.isEmpty) None else {
+      Some(s"(${left.get}) $operator (${right.get})")
+    }
+  }
+
+  private def unaryLogicalOperatorFilter(schema: StructType, childFilter: Filter, operator: String): Option[String] = {
+    val child = buildFilterExpression(schema, childFilter)
+
+    if (child.isEmpty) None else {
+      Some(s"$operator(${child.get})")
+    }
+  }
+
+  private  def stringOperatorFilter(schema: StructType, attr: String, value: String, operator: String): Option[String] = {
+    // Will return 'None' if 'attr' is not part of the 'schema'
+    getType(schema, attr).map {
+      _ => s"""$attr $operator '$value'"""
+    }
+  }
+
+  private def toStringList(values: Array[Any], dataType: DataType): String = {
+    val combined = values.map(value => format(value, dataType)).mkString(", ")
+    if (combined.isEmpty) "" else combined
```

Comment:
I think if values is empty then the mkString will return just an empty string, so you don't need this check and you can just write: values.map(value => format(value, dataType)).mkString(", ")